### PR TITLE
WIP: Write gRPC Consumer

### PR DESCRIPTION
**Pull request by ahessert**
Moving forward with the python consumer after the streaming fast [response](https://discord.com/channels/666749063386890256/820011680842907761/963902252391747614) to our data processing question.

> So, there is different possibilities depending on what you need.
Scaling through Firehose is probably the easiest when dealing with other languages than Go, you can make multiple request for different segments in parallel, for example you could fetch block by 100 000 block batches by providing "start_block_num" and "stop_block"num" and having multiple concurrent segment being processed in parallel.
Opening the files is possible, it contains block but there is forked blocks in there and consuming it correctly requires some code to handle "forks" correctly and having a consistent view of the "canonical" blocks and which have are now forked.
We have all that code in Golang to do that, bstream library is exactly what you would need. We actually exactly do this for indexing back some of our internal systems, we launched hundreds of pods in Kubernetes, each of the pod processing a segment of the chain, all in parallel so from merged blocks, we are able to re-process all the blocks of some chain in 1 hour or less (or more depending on the computation and what we do with the results).

Screen Shot 2022-04-14 at 7 31 38 PM

Roughly implementing what is drawn here.
Screen Shot 2022-04-14 at 7 30 49 PM

### CODE REVIEW

File: parquet_factory/parquet_job.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from services.gprc_service import FirehoseGRPC
+from strategies import strategy_map
+# from services.parquet_producer import make_parquet
```
Comment (SunriseLong) [inline parquet_factory/parquet_job.py:3]: unused

File: parquet_factory/job_producer.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import json
+from msilib.schema import Error
+from os import environ
+import boto3
+
+PARQUET_FACTORY_ARN = environ.get('PARQUET_FACTORY_ARN')
+AWS_REGION = environ.get('AWS_REGION')
```
Comment (SunriseLong) [inline parquet_factory/job_producer.py]: how is this populated, AWS cli?

File: parquet_factory/services/gprc_service.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+class FirehoseGRPC():
+
+    def __init__(self):
+        self.channel = grpc.insecure_channel(FIREHOSE_ENDPOINT)
+        self.stub = firehose_grpc.StreamStub(self.channel)
+    
+    @staticmethod
+    def request_args(start_block, block_count, filters):
+        request_args = {}
+        if start_block: request_args['start_block_num'] = start_block
```
Comment (SunriseLong) [inline parquet_factory/services/gprc_service.py:16]: lint, line break

**Review by SunriseLong — COMMENTED** [review]
lgtm, just needs some lint and some commented out code can be removed

**Comment by ahessert** [comment]
Merging PR. This is in good state and we can get back to smaller PRs and fast merging.

**Pull request merged**